src/main/java/jp/ac/keio/bio/fun/xitosbml/cli/CliMainImgSpatial.java

```diff
@@ -76,4 +112,71 @@ public void runCli(String imagePath, String outputPath) {
 
 	}
 
+	protected void cli(ImagePlus imager) {
+		hashDomainTypes = new HashMap<String, Integer>();
+		hashSampledValue = new HashMap<String, Integer>();
+		// imager.show();
+		trial = new TrialForImg(hashDomainTypes, hashSampledValue, imager);
+		// HashMap<String, ImagePlus> hashDomFile = trial.getDomFile();
+		// System.out.println(hashDomFile.values());
+	}
+
+	/**
+	 * Following process is performed to an image:
+	 * <ol>
+	 * <li>Interpolate an image if it is a Z-stack image (3D image) and the voxel
+	 * size of each axis (x, y and z) is not equal</li>
+	 * <li>Fill holes (blank pixels) in the image by morphology operation if
+	 * exists</li>
+	 * <li>Add a membrane between two different domains if exists</li>
+	 * </ol>
+	 *
+	 * The converted image will be generated as
+	 * {@link jp.ac.keio.bio.fun.xitosbml.image.SpatialImage}, which is a base class
+	 * for representing spatial image in XitoSBML.
+	 */
+
+	protected void computeImgTrial() {
+		Interpolator interpolator = new Interpolator();
+		HashMap<String, ImagePlus> hashDomFile = trial.getDomFile();
+		interpolator.interpolate(hashDomFile);
+		Filler fill = new Filler();
+
+		for (Entry<String, ImagePlus> e : hashDomFile.entrySet())
+			hashDomFile.put(e.getKey(), fill.fill(e.getValue()));
+
+		CreateImage creIm = new CreateImage(trial.getDomFile(), hashSampledValue);
+		spImg = new SpatialImage(hashSampledValue, hashDomainTypes, creIm.getCompoImg());
+		ImagePlus img = fill.fill(spImg);
+		spImg.setImage(img);
+		ImageBorder imgBorder = new ImageBorder(spImg);
+		spImg.updateImage(imgBorder.getStackImage());
+
+		new ImageEdit(spImg);
+
+	}
+
+	/**
+	 * Show inclusion relationship of domains as a graph.
+	 */
+	protected void showDomainStructure() {
+		spatialplugin = (SpatialModelPlugin) model.getPlugin("spatial");
+		Geometry g = spatialplugin.getGeometry();
+		new DomainStruct().show(g);
+	}
+
+	protected void print() {
+		String docStr;
+		try {
+			docStr = new TidySBMLWriter().writeSBMLToString(document);
+			System.out.println(docStr);
```

Comment:
What is the difference compared to this line? https://github.com/Medha-B/XitoSBML/blob/123be586ee57e90b68359ae1338811ab5a2c51cc/src/main/java/jp/ac/keio/bio/fun/xitosbml/cli/CliMainImgSpatial.java#L90-L99 

I think I've asked this question but is IJ.log printing to stdout? If so, it is essentially doing the same thing as this line right?

Reply:
Removed the part that was repeated. Code works fine.

Reply:
You haven't really answered my question but I'm assuming IJ.log is not printing to stdout?

Reply:
I am not sure what an IJ.log is. If it is that pop-up window with (what I am assuming) the SBML document in it, then no. If it is the SBML document itself, then that does show up on the console of Eclipse and also on the command prompt. But no separate window pops up actually.
![Screenshot (623)](https://user-images.githubusercontent.com/52010073/91058735-713ace80-e646-11ea-9229-67f688ac61d9.png)

Reply:
The only printing that ever occurs is on the console.

Reply:
Google IJ.log. No log window pops up if we are going by this https://imagej.nih.gov/ij/docs/guide/146-24.html.

Reply:
Looking at the code, IJ.log is printing in stdout if the logPanel(which is the GUI in ImageJ) is null. So I presume the SBMLDocument was printed out twice in stdout? But we remove the IJ.log part so we should be fine.

https://imagej.nih.gov/ij/developer/source/ij/IJ.java.html
```
    public static synchronized void log(String s) {
        if (s==null) return;
        if (logPanel==null && ij!=null) {
            TextWindow logWindow = new TextWindow("Log", "", 400, 250);
            logPanel = logWindow.getTextPanel();
            logPanel.setFont(new Font("SansSerif", Font.PLAIN, 16));
        }
        if (logPanel!=null) {
            if (s.startsWith("\\"))
                handleLogCommand(s);
            else {
                if (s.endsWith("\n")) {
                    if (s.equals("\n\n"))
                        s= "\n \n ";
                    else if (s.endsWith("\n\n"))
                        s = s.substring(0, s.length()-2)+"\n \n ";
                    else
                        s = s+" ";
                }
                logPanel.append(s);
            }
        } else {
            LogStream.redirectSystem(false);
            System.out.println(s);
        }
    }
```